### PR TITLE
[#17] Add more map-like functions

### LINKED ISSUE #17: Add more map-like functions
`member` `delete` `singleton`

### PR DESCRIPTION
**Pull request by ChShersh**
Resolves #17 

Thanks @int-index for idea of this `KindOf` trick!

### CODE REVIEW

File: internal/Data/TypeRep/CacheMap.hs [hunk 1]
```diff
@@ -70,32 +73,62 @@ fromAny = unsafeCoerce
 empty :: TypeRepMap f
 empty = TypeRepMap mempty mempty mempty
 
+-- | Creates structure of size 1 from single element.
```
Comment (vrom911) [inline internal/Data/TypeRep/CacheMap.hs]: Probably it's better to say *Construct a 'TypeRepMap' with a single element.*

File: typerep-map.cabal [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+version:             1.0
 description:         Efficient implementation of data type with types as keys
 homepage:            https://github.com/vrom911/typerep-map
 bug-reports:         https://github.com/vrom911/typerep-map/issues
 license:             MIT
 license-file:        LICENSE
 author:              Veronika Romashkina
 maintainer:          vrom911@gmail.com
-copyright:           2017 Veronika Romashkina
+copyright:           2017-2018 Veronika Romashkina
```
Comment (vrom911) [inline typerep-map.cabal]: As you're on it, could you please also change the links, names to Kowainik and add Vladislav as well?

File: internal/Data/TypeRep/CacheMap.hs [hunk 3]
```diff
@@ -70,32 +73,62 @@ fromAny = unsafeCoerce
...
+>>> member @Bool trMap
+False
+>>> member @Char trMap
+True
+-}
+delete :: forall a (f :: KindOf a -> Type) . Typeable a => TypeRepMap f -> TypeRepMap f
+delete = fromListPairs . deleteByFst (typeFp @a) . toPairList
+{-# INLINE delete #-}
+
+type KindOf (a :: k) = k
```
Comment (vrom911) [inline internal/Data/TypeRep/CacheMap.hs]: 😮

File: internal/Data/TypeRep/CacheMap.hs [hunk 4]
```diff
@@ -70,32 +73,63 @@ fromAny = unsafeCoerce
 empty :: TypeRepMap f
 empty = TypeRepMap mempty mempty mempty
 
+-- | Construct a 'TypeRepMap' with a single element.
+one :: forall a f . Typeable a => f a -> TypeRepMap f
```
Comment (int-index) [inline internal/Data/TypeRep/CacheMap.hs]: There are many possible names for this function: `one`, `only`, `single`, `singleton`, etc. I think the choice should be made in favor of the most common one, `singleton` - it's used by both `containers` and `dependent-map`.

Comment (ChShersh) [reply internal/Data/TypeRep/CacheMap.hs]: @int-index I known that `singleton` is more common name. But I'd like to have shorter name. Though, it's minor, can be renamed.

Comment (int-index) [reply internal/Data/TypeRep/CacheMap.hs]: I don't like the name `singleton` anyway, so use your judgement. `one` is equally sensible, just not what I'm used to.

**Pull request merged**